src/commands/commerce/extension/unmap.ts

```diff
@@ -0,0 +1,105 @@
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { flags, SfdxCommand } from '@salesforce/command';
+import { Messages, Org, SfdxError } from '@salesforce/core';
+import { QueryResult } from '@mshanemc/plugin-helpers/dist/typeDefs';
+import { forceDataSoql, forceDataRecordDelete } from '../../../lib/utils/sfdx/forceDataSoql';
+import { StatusFileManager } from '../../../lib/utils/statusFileManager';
+import { Result } from '../../../lib/utils/jsonUtils';
+
+Messages.importMessagesDirectory(__dirname);
+
+const TOPIC = 'extension';
+const CMD = `commerce:${TOPIC}:unmap`;
+const msgs = Messages.loadMessages('@salesforce/commerce', 'store');
+
+export class UnMapExtension extends SfdxCommand {
+    public static readonly requiresUsername = true;
+    public static description = msgs.getMessage('extension.unmap.cmdDescription');
+    public static example = [
+        `sfdx ${CMD} --registered-extension-name test-extension-name --store-name test-store-name `,
+        `sfdx ${CMD} --registered-extension-name test-extension-name --store-id test-store-id `,
+    ];
+    public static flagsConfig = {
+        'registered-extension-name': flags.string({
+            char: 'r',
+            description: msgs.getMessage('extension.unmap.regExtensionNameFlagDescription'),
+        }),
+        'store-name': flags.string({
+            char: 'n',
+            description: msgs.getMessage('extension.unmap.StoreNameFlagDescription'),
+        }),
+        'store-id': flags.string({
+            char: 'i',
+            description: msgs.getMessage('extension.unmap.storeId'),
+        }),
+    };
+    public org: Org;
+    public statusFileManager: StatusFileManager;
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public async run(): Promise<void> {
+        this.unmapRecord(
+            this.flags['registered-extension-name'],
+            this.flags['store-name'],
+            this.flags['store-id'],
+            this.org.getUsername()
+        );
+    }
+
+    public unmapRecord(extensionName: string, storeName: string, storeId: string, userName: string): void {
+        if (storeName === undefined && storeId === undefined) {
+            throw new SfdxError(msgs.getMessage('extension.unmap.undefinedName'));
+        }
+        const storeid = this.validateStoreId(storeName, storeId, userName);
+        const name = forceDataSoql(`SELECT Name FROM WebStore WHERE Id='${storeid}' LIMIT 1`).result.records[0].Name;
+        const integration = 'Extension'.concat('__' as string).concat(extensionName);
+        try {
+            const deleteId = forceDataSoql(
+                `SELECT Id FROM StoreIntegratedService WHERE Integration='${integration}' AND storeid='${storeid}'`,
+                userName
+            ).result.records[0].Id;
+            forceDataRecordDelete('StoreIntegratedService', deleteId, this.org.getUsername());
```

Comment:
this also looks like a candidate to create a method for which map.ts and unmap.ts call